### PR TITLE
Allow custom user agent

### PR DESCRIPTION
**Pull request by pauldendulk**
Why: Because the default user agents can get blocked by osm

### CODE REVIEW

**Comment by pauldendulk** [comment]
Context: The layer sends a user-agent at every request. The previous user-agent used in BruTile got blocked by osm at some point. This blocking probably happens because someone used the BruTile library to bulk load tiles. This is not allowed by the osm service. As a consequence every app that is using that user-agent will not receive tiles anymore. This could also happen to the default user-agent of Mapsui. So it should become common practice to specify your own user-agent if you create an app.

File: Mapsui/Utilities/OpenStreetMap.cs [hunk 1]
```diff
@@ -14,12 +15,12 @@ public static TileLayer CreateTileLayer()
             return new TileLayer(CreateTileSource()) { Name = "OpenStreetMap" };
```
Comment (inforithmics) [inline Mapsui/Utilities/OpenStreetMap.cs:15]: Shouldn't there be a parameter here with the userAgent.
or the Alternative would be that the DefaultUserAgent is not a constant but a Static Property that can be set from outside

File: Mapsui/Utilities/OpenStreetMap.cs [hunk 2]
```diff
@@ -14,12 +15,12 @@ public static TileLayer CreateTileLayer()
...
         }
 
-        private static HttpTileSource CreateTileSource()
+        private static HttpTileSource CreateTileSource(string userAgent = DefaultUserAgent)
```
Comment (inforithmics) [inline Mapsui/Utilities/OpenStreetMap.cs:18]: I don't think that this can be called externally because it is private static ..

Comment (pauldendulk) [reply Mapsui/Utilities/OpenStreetMap.cs]: Ah, you are right. Too hurried. I pushed an update.

**Review by inforithmics — APPROVED** [review]

**Comment by janusw** [comment]
I'm also ok with the changes here. I don't use the `OpenStreetMap` class, though, but directly rely on the BruTile functionality for OSM, where a custom user agent is already supported:
```
ILayer l = new TileLayer(KnownTileSources.Create(KnownTileSource.OpenStreetMap, userAgent: "myUA", minZoomLevel: minZL, maxZoomLevel: maxZL));
```
Feel free to merge this, Paul!

**Pull request merged**